lib/mayaUsd/undo/OpUndoInfo.h

```diff
@@ -0,0 +1,148 @@
+//
+// Copyright 2021 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifndef MAYAUSD_UNDO_OP_UNDO_INFO_H
+#define MAYAUSD_UNDO_OP_UNDO_INFO_H
+
+#include <mayaUsd/base/api.h>
+
+#include <maya/MObjectHandle.h>
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+namespace MAYAUSD_NS_DEF {
+
+//------------------------------------------------------------------------------
+// OpUndoItem
+//------------------------------------------------------------------------------
+
+/// \class OpUndoItem
+/// \brief Record data needed to undo or redo a single undo sub-operation.
+///
+/// See OpUndoItems.h for concrete implementations.
+
+class OpUndoItem
```

Comment:
The edit-as-maya nd merge-to-usd code needs to call commands, Maya API, Python code, etc. Some of these are not fully supported in MDGModifier. For example, it can run Python code, but we cannot provide the 'undo' Python code.

So this undo item interface gets implemented by classes in OpUndoItems.h to support all the need to the prim updater.